crates/noirc_frontend/src/hir/def_collector/dc_crate.rs

```diff
@@ -227,9 +227,14 @@ fn collect_impls(
                     }
                 }
             } else if typ != Type::Error {
-                let span = *span;
-                let error = DefCollectorErrorKind::NonStructTypeInImpl { span };
-                errors.push(error.into_file_diagnostic(unresolved.file_id))
+                if true
```

Comment:
what is this line meant to be doing?

Reply:
Ah, that was leftover from me trying to find a way to restrict primitive impls to only appearing in the stdlib. I could not find a good method in the end since I could not find the crate id of the stdlib, only the local crate.

Reply:
Could we refactor this, If I'm reading it correctly, it says `if true {} else {// do something else}`

Reply:
Ah, yes forgot about this block in the last commit. I added a somewhat better check to check that this isn't the local crate at least since we have no way to check if it is the stdlib or not. This should at least prevent most misuses but not all.

Reply:
Yeah I think we can open up an issue and tackle this in a later PR. We may need to special case the stdlib in the compiler